src/System.Windows.Forms/src/System/Windows/Forms/Control.cs

```diff
@@ -4694,6 +4694,13 @@ protected override void Dispose(bool disposing)
             }
         }
 
+        // Unsubscribes from the Disposed event of the ContextMenuStrip.
+        ContextMenuStrip? menu = ContextMenuStrip;
```

Comment:
```suggestion
       if (ContextMenuStrip is  ContextMenuStrip menu)
```
you can use pattern matching here.